File: Fcc_Javascript_Cert_Answers/ceaser_cipher.py
# Function equivalent to passing code for ceaser cipher challenge for fcc js certification


def rot13(s):
    alphabet = list("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
    punct_list = [" ", ",", "!", "?", "."]
    decoded_s = ""
    for letter in s:
        if letter in punct_list:
            decoded_s += letter
        elif 0 <= alphabet.index(letter) <= 12:
            decoded_ind = alphabet.index(letter) + 13
            decoded_s += alphabet[decoded_ind]
        else:
            decoded_ind = alphabet.index(letter) - 13
            decoded_s += alphabet[decoded_ind]
    return decoded_s


print(rot13("SERR PBQR PNZC!"))

# Note: whereas javascript can take a string like alphabet and turn it into a list
# using split() with an empty seperator, Python cannot. Nor does it need to, since the
# same effect can be achieved using list().

# the only other new difference here -- alphabet.index instead of alphabet.indexOf()

File: Fcc_Javascript_Cert_Answers/ceaser_cipher.js
// Write a function which takes a ROT13 encoded string as input and returns a decoded string.
function rot13(str) {
  let alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("");
  let punctArray = [" ", ",", "!", "?", "."]
  let decodedStr = "";
  var decodedIndex = 0;
  for(let index in str) {
    if (punctArray.includes(str[index])) {
      decodedStr += str[index];
    }
    if ((alphabet.indexOf(str[index])) <= 12 && (alphabet.indexOf(str[index])) >= 0) {
      decodedIndex = (alphabet.indexOf(str[index])) + 13;
      decodedStr += alphabet[decodedIndex];
    } else if ((alphabet.indexOf(str[index])) >= 13 && (alphabet.indexOf(str[index])) <= 25) {
      decodedIndex = (alphabet.indexOf(str[index])) - 13;
      decodedStr += alphabet[decodedIndex];
    } 
  } 
    return decodedStr;
  }
  
console.log(rot13("SERR PBQR PNZC!"));
